diff -ru L0/FortressBuiltin.fsi T1/FortressBuiltin.fsi
--- L0/FortressBuiltin.fsi	2026-09-27 16:24:17.114927490 +0000
+++ T1/FortressBuiltin.fsi	2026-09-27 16:31:44.549186763 +0000
@@ -27,7 +27,7 @@
     generally easier to work with, and the boilerplate packing and
     unpacking of values is done for you.
 **)
-builtinPrimitive[\T\](javaClass:String):T
+builtinPrimitive[\T extends Object\](javaClass:String):T
 
 trait Object extends Any
     getter ilkName(): String
diff -ru L0/FortressBuiltin.fss T1/FortressBuiltin.fss
--- L0/FortressBuiltin.fss	2026-09-27 16:24:17.115143504 +0000
+++ T1/FortressBuiltin.fss	2026-09-27 16:31:44.549435947 +0000
@@ -31,7 +31,7 @@
     generally easier to work with, and the boilerplate packing and
     unpacking of values is done for you.
 **)
-builtinPrimitive[\T\](javaClass:String):T =
+builtinPrimitive[\T extends Object\](javaClass:String):T =
   fail ("Implementation should not invoke builtinPrimitive " javaClass)
 
 trait Object extends Any
diff -ru L0/FortressLibrary.fsi T1/FortressLibrary.fsi
--- L0/FortressLibrary.fsi	2026-09-27 16:24:17.112012920 +0000
+++ T1/FortressLibrary.fsi	2026-09-27 16:31:44.621001946 +0000
@@ -34,7 +34,7 @@
 (* Function composition *)
 opr COMPOSE[\A,B,C\](f: B->C, g: A->B): A->C
 
-fail[\T\](s:String):T
+fail[\T extends Object\](s:String):T
 
 (************************************************************
 * \subsection*{Control over locality and location}
@@ -206,8 +206,8 @@
 trait StandardMinMax[\T extends StandardMinMax[\T\]\]
         extends { StandardMin[\T\], StandardMax[\T\] }
     abstract opr MINMAX(self, other:T): (T,T)
-    opr MIN(self, other:T): (T,T)
-    opr MAX(self, other:T): (T,T)
+    opr MIN(self, other:T): T
+    opr MAX(self, other:T): T
 end
 
 (** StandardTotalOrder is the usual total order using %<%,%>%,%<=%,%>=%,%=%, and %CMP%.
@@ -2194,34 +2194,34 @@
 trait StridedFullRange[\I\] extends FullRange[\I\] end
 
 (** The %#% and %:% operators serve as factories for parallel ranges. **)
-opr #[\I extends AnyIntegral\](lo:I, ex:I): CompactFullRange[\I\]
-opr #[\I extends AnyIntegral, J extends AnyIntegral\]
+opr #[\I extends Integral[\I\]\](lo:I, ex:I): CompactFullRange[\I\]
+opr #[\I extends Integral[\I\], J extends Integral[\J\]\]
      (lo:(I,J), ex:(I,J)): CompactFullRange[\(I,J)\]
-opr #[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+opr #[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
      (lo:(I,J,K), ex:(I,J,K)): CompactFullRange[\(I,J,K)\]
 
-opr :[\I extends AnyIntegral\](lo:I, hi:I): CompactFullRange[\I\]
-opr :[\I extends AnyIntegral, J extends AnyIntegral\]
+opr :[\I extends Integral[\I\]\](lo:I, hi:I): CompactFullRange[\I\]
+opr :[\I extends Integral[\I\], J extends Integral[\J\]\]
      (lo:(I,J), hi:(I,J)): CompactFullRange[\(I,J)\]
-opr :[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+opr :[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
      (lo:(I,J,K), hi:(I,J,K)): CompactFullRange[\(I,J,K)\]
 
 (** Factories for incomplete ranges. **)
-opr (x:I)#[\I extends AnyIntegral\] : LeftRange[\I\]
-opr (p:(I,J))#[\I extends AnyIntegral, J extends AnyIntegral\] : LeftRange[\(I,J)\]
-opr (t:(I,J,K))#[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\] :
+opr (x:I)#[\I extends Integral[\I\]\] : LeftRange[\I\]
+opr (p:(I,J))#[\I extends Integral[\I\], J extends Integral[\J\]\] : LeftRange[\(I,J)\]
+opr (t:(I,J,K))#[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\] :
          LeftRange[\(I,J,K)\]
 
 opr (x:I):[\I\] : LeftRange[\I\]
 
-opr #[\I extends AnyIntegral\](x:I) : ExtentRange[\I\]
-opr #[\I extends AnyIntegral, J extends AnyIntegral\](xy:(I,J)) : ExtentRange[\(I,J)\]
-opr #[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](xyz:(I,J,K)) :
+opr #[\I extends Integral[\I\]\](x:I) : ExtentRange[\I\]
+opr #[\I extends Integral[\I\], J extends Integral[\J\]\](xy:(I,J)) : ExtentRange[\(I,J)\]
+opr #[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](xyz:(I,J,K)) :
          ExtentRange[\(I,J,K)\]
 
-opr :[\I extends AnyIntegral\](x:I) : RightRange[\I\]
-opr :[\I extends AnyIntegral, J extends AnyIntegral\](xy:(I,J)) : RightRange[\(I,J)\]
-opr :[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](xyz:(I,J,K)) :
+opr :[\I extends Integral[\I\]\](x:I) : RightRange[\I\]
+opr :[\I extends Integral[\I\], J extends Integral[\J\]\](xy:(I,J)) : RightRange[\(I,J)\]
+opr :[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](xyz:(I,J,K)) :
          RightRange[\(I,J,K)\]
 
 (* Actually want:
@@ -2246,9 +2246,9 @@
 
 opr (l:I)::[\I\] : LeftRange[\I\]
 
-opr ::[\I extends AnyIntegral\](s:I) : OpenRange[\I\]
-opr ::[\I extends AnyIntegral, J extends AnyIntegral\](ij:(I,J)) : OpenRange[\(I,J)\]
-opr ::[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+opr ::[\I extends Integral[\I\]\](s:I) : OpenRange[\I\]
+opr ::[\I extends Integral[\I\], J extends Integral[\J\]\](ij:(I,J)) : OpenRange[\(I,J)\]
+opr ::[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
       (ijk:(I,J,K)) : OpenRange[\(I,J,K)\]
 
 opr ::[\I\](l:I,s:I): LeftRange[\I\]
diff -ru L0/FortressLibrary.fss T1/FortressLibrary.fss
--- L0/FortressLibrary.fss	2026-09-27 16:24:17.113515819 +0000
+++ T1/FortressLibrary.fss	2026-09-27 16:31:44.652785320 +0000
@@ -51,7 +51,7 @@
 (* Function composition *)
 opr COMPOSE[\A,B,C\](f: B->C, g: A->B): A->C = fn (a:A): C => f(g(a))
 
-fail[\T\](s:String):T = do
+fail[\T extends Object\](s:String):T = do
     errorPrintln("FAIL: " s)
     throw FailCalled(s)
   end
@@ -257,8 +257,8 @@
 trait StandardMinMax[\T extends StandardMinMax[\T\]\]
         extends { StandardMin[\T\], StandardMax[\T\] }
     opr MINMAX(self, other:T): (T,T)
-    opr MIN(self, other:T): (T,T) = do (r,_) = self MINMAX other; r end
-    opr MAX(self, other:T): (T,T) = do (_,r) = self MINMAX other; r end
+    opr MIN(self, other:T): T = do (r,_) = self MINMAX other; r end
+    opr MAX(self, other:T): T = do (_,r) = self MINMAX other; r end
 end
 
 (** StandardTotalOrder is the usual total order using <,>,<=,>=,=, and
@@ -3885,54 +3885,54 @@
 trait StridedFullRange[\I\] extends FullRange[\I\] end
 
 (** The # and : operators serve as factories for parallel ranges. **)
-opr #[\I extends AnyIntegral\](lo:I, ex:I): CompactFullParScalarRange[\I\] = sized1Range(0 asif ZZ32,lo,ex)
-opr #[\I extends AnyIntegral, J extends AnyIntegral\]
+opr #[\I extends Integral[\I\]\](lo:I, ex:I): CompactFullParScalarRange[\I\] = sized1Range(lo, lo, ex)
+opr #[\I extends Integral[\I\], J extends Integral[\J\]\]
      (lo:(I,J), ex:(I,J)): CompactFullRange2D[\I,J\] =
-    do (l1,l2)=lo; (x1,x2)=ex; sized2Range(0 asif ZZ32,0 asif ZZ32,l1,l2,x1,x2) end
-opr #[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+    do (l1,l2)=lo; (x1,x2)=ex; sized2Range(l1, l2, l1, l2, x1, x2) end
+opr #[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
      (lo:(I,J,K), ex:(I,J,K)): CompactFullRange3D[\I,J,K\] =
     do (l1,l2,l3)=lo; (x1,x2,x3)=ex
-       sized3Range(0 asif ZZ32,0 asif ZZ32,0 asif ZZ32,l1,l2,l3,x1,x2,x3)
+       sized3Range(l1, l2, l3, l1, l2, l3, x1, x2, x3)
     end
 
-opr :[\I extends AnyIntegral\](lo:I, hi:I): CompactFullParScalarRange[\I\] =
-        bounded1Range(0 asif ZZ32,lo,hi)
-opr :[\I extends AnyIntegral, J extends AnyIntegral\]
+opr :[\I extends Integral[\I\]\](lo:I, hi:I): CompactFullParScalarRange[\I\] =
+        bounded1Range(lo, lo, hi)
+opr :[\I extends Integral[\I\], J extends Integral[\J\]\]
      (lo:(I,J), hi:(I,J)): CompactFullRange2D[\I,J\] =
-    do (l1,l2)=lo; (h1,h2)=hi; bounded2Range(0 asif ZZ32,0 asif ZZ32,l1,l2,h1,h2) end
-opr :[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+    do (l1,l2)=lo; (h1,h2)=hi; bounded2Range(l1, l2, l1, l2, h1, h2) end
+opr :[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
      (lo:(I,J,K), hi:(I,J,K)): CompactFullRange3D[\I,J,K\] = do
     (l1,l2,l3)=lo; (h1,h2,h3)=hi
-    bounded3Range(0 asif ZZ32,0 asif ZZ32,0 asif ZZ32,l1,l2,l3,h1,h2,h3)
+    bounded3Range(l1, l2, l3, l1, l2, l3, h1, h2, h3)
   end
 
 (** Factories for incomplete ranges **)
-opr (x:I)#[\I extends AnyIntegral\] : LeftRange[\I\] = left1Range(0 asif ZZ32, x)
-opr (p:(I,J))#[\I extends AnyIntegral, J extends AnyIntegral\] : LeftRange[\(I,J)\] = do
+opr (x:I)#[\I extends Integral[\I\]\] : LeftRange[\I\] = left1Range(x, x)
+opr (p:(I,J))#[\I extends Integral[\I\], J extends Integral[\J\]\] : LeftRange[\(I,J)\] = do
     (x, y) = p
-    left2Range(0 asif ZZ32,0 asif ZZ32, x,y)
+    left2Range(x, y, x, y)
   end
-opr (t:(I,J,K))#[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\] :
+opr (t:(I,J,K))#[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\] :
          LeftRange[\(I,J,K)\] = do
     (x, y, z) = t
-    left3Range(0 asif ZZ32,0 asif ZZ32,0 asif ZZ32, x,y,z)
+    left3Range(x, y, z, x, y, z)
   end
 
 opr (x:I):[\I\] : LeftRange[\I\] = x#
 
-opr #[\I extends AnyIntegral\](x:I) : ExtentRange[\I\] = extent1Range(0 asif ZZ32, x)
-opr #[\I extends AnyIntegral, J extends AnyIntegral\](xy:(I,J)) : ExtentRange[\(I,J)\] =
-    do (x,y) = xy; extent2Range(0 asif ZZ32,0 asif ZZ32, x,y) end
-opr #[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](xyz:(I,J,K)) :
+opr #[\I extends Integral[\I\]\](x:I) : ExtentRange[\I\] = extent1Range(x, x)
+opr #[\I extends Integral[\I\], J extends Integral[\J\]\](xy:(I,J)) : ExtentRange[\(I,J)\] =
+    do (x,y) = xy; extent2Range(x, y, x, y) end
+opr #[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](xyz:(I,J,K)) :
          ExtentRange[\(I,J,K)\] =
-    do (x,y,z) = xyz; extent3Range(0 asif ZZ32,0 asif ZZ32,0 asif ZZ32, x,y,z) end
+    do (x,y,z) = xyz; extent3Range(x, y, z, x, y, z) end
 
-opr :[\I extends AnyIntegral\](x:I) : RightRange[\I\] = right1Range(0 asif ZZ32, x)
-opr :[\I extends AnyIntegral, J extends AnyIntegral\](xy:(I,J)) : RightRange[\(I,J)\] =
-    do (x,y) = xy; right2Range(0 asif ZZ32,0 asif ZZ32, x,y) end
-opr :[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](xyz:(I,J,K)) :
+opr :[\I extends Integral[\I\]\](x:I) : RightRange[\I\] = right1Range(x, x)
+opr :[\I extends Integral[\I\], J extends Integral[\J\]\](xy:(I,J)) : RightRange[\(I,J)\] =
+    do (x,y) = xy; right2Range(x, y, x, y) end
+opr :[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](xyz:(I,J,K)) :
          RightRange[\(I,J,K)\] =
-    do (x,y,z) = xyz; right3Range(0 asif ZZ32,0 asif ZZ32,0 asif ZZ32, x,y,z) end
+    do (x,y,z) = xyz; right3Range(x, y, z, x, y, z) end
 
 (* Actually want for above:
 opr (x:I)#[\I\] : LeftRange[\I\] = LeftRange[\I\](x)
@@ -3956,12 +3956,12 @@
 
 opr (l:I)::[\I\] : LeftRange[\I\] = l#
 
-opr ::[\I extends AnyIntegral\](s:I) : OpenRange[\I\] = open1Range(0 asif ZZ32, s)
-opr ::[\I extends AnyIntegral, J extends AnyIntegral\](ij:(I,J)) : OpenRange[\(I,J)\] =
-    do (i,j) = ij; open2Range(0 asif ZZ32, 0 asif ZZ32, i,j) end
-opr ::[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+opr ::[\I extends Integral[\I\]\](s:I) : OpenRange[\I\] = open1Range(s, s)
+opr ::[\I extends Integral[\I\], J extends Integral[\J\]\](ij:(I,J)) : OpenRange[\(I,J)\] =
+    do (i,j) = ij; open2Range(i, j, i, j) end
+opr ::[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
       (ijk:(I,J,K)) : OpenRange[\(I,J,K)\] =
-    do (i,j,k) = ijk; open3Range(0 asif ZZ32, 0 asif ZZ32, 0 asif ZZ32, i,j,k) end
+    do (i,j,k) = ijk; open3Range(i, j, k, i, j, k) end
 
 opr ::[\I\](l:I,s:I): LeftRange[\I\] = (l:):s
 
diff -ru L0/List.fsi T1/List.fsi
--- L0/List.fsi	2026-09-27 16:24:17.112260304 +0000
+++ T1/List.fsi	2026-09-27 16:31:44.558440897 +0000
@@ -106,7 +106,7 @@
 (** Vararg factory for lists; provides aggregate list constants: *)
 opr <|[\E\] xs: E... |>: List[\E\]
 (** List comprehensions: *)
-opr BIG <|[\T\]|>:Comprehension[\T,List[\T\],List[\T\],List[\T\]\]
+opr BIG <|[\T extends Object\]|>:Comprehension[\T,List[\T\],List[\T\],List[\T\]\]
 opr BIG <|[\T\] g:Generator[\T\]|>:List[\T\]
 
 opr BIG CONCAT[\T\](): BigReduction[\List[\T\],List[\T\]\]
diff -ru L0/List.fss T1/List.fss
--- L0/List.fss	2026-09-27 16:24:17.113829989 +0000
+++ T1/List.fss	2026-09-27 16:31:44.558635667 +0000
@@ -174,7 +174,7 @@
 
 (** Vararg factory for lists; provides aggregate list constants **)
 opr <|[\E\] xs: E... |>: List[\E\] = list(xs)
-opr BIG <|[\T\]|>: Comprehension[\T,List[\T\],AnyCovColl,AnyCovColl\] =
+opr BIG <|[\T extends Object\]|>: Comprehension[\T,List[\T\],AnyCovColl,AnyCovColl\] =
     covariantCompr[\T,List[\T\]\](fn (cc) => fromImmutableArray(cc.toImmutableArray()))
 opr BIG <|[\T\] g:Generator[\T\]|>:List[\T\] =
     __bigOperatorSugar[\T,List[\T\],AnyCovColl,AnyCovColl\](BIG <|[\T\]|>(), g)
diff -ru L0/RangeInternals.fsi T1/RangeInternals.fsi
--- L0/RangeInternals.fsi	2026-09-27 16:24:17.112545515 +0000
+++ T1/RangeInternals.fsi	2026-09-27 16:31:44.622863595 +0000
@@ -591,23 +591,23 @@
 bounded3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](_: I, _: J, _: K, l1: I, l2: J, l3: K, hi1: I, hi2: J,
         hi3: K): CompactFullRange3D[\I, J, K\]
 
-left1Range[\I extends AnyIntegral\](_: I, x: I): LeftRange[\I\]
+left1Range[\I extends Integral[\I\]\](_: I, x: I): LeftRange[\I\]
 
-left2Range[\I extends AnyIntegral, J extends AnyIntegral\](_: I, _: J, x: I, y: J): LeftRange[\(I, J)\]
+left2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_: I, _: J, x: I, y: J): LeftRange[\(I, J)\]
 
-left3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](_: I, _: J, _: K, x: I, y: J, z: K): LeftRange[\(I, J, K)\]
+left3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](_: I, _: J, _: K, x: I, y: J, z: K): LeftRange[\(I, J, K)\]
 
-extent1Range[\I extends AnyIntegral\](_: I, x: I): ExtentRange[\I\]
+extent1Range[\I extends Integral[\I\]\](_: I, x: I): ExtentRange[\I\]
 
-extent2Range[\I extends AnyIntegral, J extends AnyIntegral\](_: I, _: J, x: I, y: J): ExtentRange[\(I, J)\]
+extent2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_: I, _: J, x: I, y: J): ExtentRange[\(I, J)\]
 
-extent3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](_: I, _: J, _: K, x: I, y: J, z: K): ExtentRange[\(I, J, K)\]
+extent3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](_: I, _: J, _: K, x: I, y: J, z: K): ExtentRange[\(I, J, K)\]
 
-right1Range[\I extends AnyIntegral\](_: I, x: I): RightRange[\I\]
+right1Range[\I extends Integral[\I\]\](_: I, x: I): RightRange[\I\]
 
-right2Range[\I extends AnyIntegral, J extends AnyIntegral\](_: I, _: J, x: I, y: J): RightRange[\(I, J)\]
+right2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_: I, _: J, x: I, y: J): RightRange[\(I, J)\]
 
-right3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](_: I, _: J, _: K, x: I, y: J, z: K): RightRange[\(I, J, K)\]
+right3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](_: I, _: J, _: K, x: I, y: J, z: K): RightRange[\(I, J, K)\]
 
 openRangeHelper[\I extends Integral[\I\]\](_: ()->I): OpenScalarRange[\I\]
 
@@ -615,10 +615,10 @@
 
 openRangeHelper[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](_: ()->(I, J, K)): OpenRange3D[\I, J, K\]
 
-open1Range[\I extends AnyIntegral\](_: I, x: I): OpenRange[\I\]
+open1Range[\I extends Integral[\I\]\](_: I, x: I): OpenRange[\I\]
 
-open2Range[\I extends AnyIntegral, J extends AnyIntegral\](_: I, _: J, x: I, y: J): OpenRange[\(I, J)\]
+open2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_: I, _: J, x: I, y: J): OpenRange[\(I, J)\]
 
-open3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\](_: I, _: J, _: K, x: I, y: J, z: K): OpenRange[\(I, J, K)\]
+open3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\](_: I, _: J, _: K, x: I, y: J, z: K): OpenRange[\(I, J, K)\]
 
 end
diff -ru L0/RangeInternals.fss T1/RangeInternals.fss
--- L0/RangeInternals.fss	2026-09-27 16:24:17.114232753 +0000
+++ T1/RangeInternals.fss	2026-09-27 16:31:44.622147491 +0000
@@ -678,7 +678,7 @@
         end
 end
 
-object RightScalarRange[\I\](r:I, str:I)
+object RightScalarRange[\I extends Integral[\I\]\](r:I, str:I)
         extends { ScalarRangeWithRight[\I\], PartialScalarRange[\I\], RightRange[\I\] }
     getter stride(): I = str
     getter right(): Just[\I\] = Just[\I\](r)
@@ -1415,44 +1415,44 @@
         StridedFullRange3D[\I,J,K\](l_i,l_j,l_k,r_i,r_j,r_k,str_i,str_j,str_k).check()
     end
 
-emptyScalarRange[\I\]() : FullScalarRange[\I\] = CompactFullParScalarRange[\I\](0,-1)
+emptyScalarRange[\I extends Integral[\I\]\]() : FullScalarRange[\I\] = CompactFullParScalarRange[\I\](0,-1)
 
 (** Helpers for # to get the type instantiation "right".  We pass in bogus ZZ32's to
     ensure that the result type is at least ZZ32. **)
-sized1Range[\I extends AnyIntegral\](_:I,lo:I,ex:I): CompactFullParScalarRange[\I\] =
+sized1Range[\I extends Integral[\I\]\](_:I,lo:I,ex:I): CompactFullParScalarRange[\I\] =
     CompactFullParScalarRange[\I\](lo,lo+ex-1)
-sized2Range[\I extends AnyIntegral, J extends AnyIntegral\]
+sized2Range[\I extends Integral[\I\], J extends Integral[\J\]\]
            (_:I,_:J,l1:I,l2:J,ex1:I,ex2:J): CompactFullRange2D[\I,J\] =
     CompactFullRange2D[\I,J\](l1,l2,l1+ex1-1,l2+ex2-1)
-sized3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+sized3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
      (_:I,_:J,_:K,l1:I,l2:J,l3:K,ex1:I,ex2:J,ex3:K): CompactFullRange3D[\I,J,K\] =
     CompactFullRange3D[\I,J,K\](l1,l2,l3,l1+ex1-1,l2+ex2-1,l3+ex3-1)
 
 (** Helpers for : to get the type instantiation "right".  We pass in bogus ZZ32's to
     ensure that the result type is at least ZZ32.  **)
-bounded1Range[\I extends AnyIntegral\](_:I,lo:I,hi:I): CompactFullParScalarRange[\I\] = CompactFullParScalarRange[\I\](lo,hi)
-bounded2Range[\I extends AnyIntegral, J extends AnyIntegral\]
+bounded1Range[\I extends Integral[\I\]\](_:I,lo:I,hi:I): CompactFullParScalarRange[\I\] = CompactFullParScalarRange[\I\](lo,hi)
+bounded2Range[\I extends Integral[\I\], J extends Integral[\J\]\]
            (_:I,_:J,l1:I,l2:J,hi1:I,hi2:J): CompactFullRange2D[\I,J\] =
     CompactFullRange2D[\I,J\](l1,l2,hi1,hi2)
-bounded3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+bounded3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
      (_:I,_:J,_:K,l1:I,l2:J,l3:K,hi1:I,hi2:J,hi3:K): CompactFullRange3D[\I,J,K\] =
     CompactFullRange3D[\I,J,K\](l1,l2,l3,hi1,hi2,hi3)
 
-left1Range[\I extends AnyIntegral\](_:I,x:I):LeftRange[\I\] = LeftScalarRange[\I\](x,1)
-left2Range[\I extends AnyIntegral, J extends AnyIntegral\](_:I,_:J,x:I,y:J):
+left1Range[\I extends Integral[\I\]\](_:I,x:I):LeftRange[\I\] = LeftScalarRange[\I\](x,1)
+left2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_:I,_:J,x:I,y:J):
     LeftRange[\(I,J)\] = LeftRange2D[\I,J\](x,y,1,1)
-left3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+left3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
            (_:I,_:J,_:K,x:I,y:J,z:K): LeftRange[\(I,J,K)\] =
     LeftRange3D[\I,J,K\](x,y,z,1,1,1)
 
-extent1Range[\I extends AnyIntegral\](_:I,x:I):ExtentRange[\I\] =
+extent1Range[\I extends Integral[\I\]\](_:I,x:I):ExtentRange[\I\] =
     if x=0 then
         xx = x-x
         CompactFullParScalarRange[\I\](xx,xx-1)
     else
         ExtentScalarRange[\I\](x,1)
     end
-extent2Range[\I extends AnyIntegral, J extends AnyIntegral\](_:I,_:J,x:I,y:J):
+extent2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_:I,_:J,x:I,y:J):
         ExtentRange[\(I,J)\] =
     if x=0 OR y=0 then
         xx = x-x
@@ -1461,7 +1461,7 @@
     else
         ExtentRange2D[\I,J\](x,y,1,1)
     end
-extent3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+extent3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
             (_:I,_:J,_:K,x:I,y:J,z:K): ExtentRange[\(I,J,K)\] =
     if x=0 OR y=0 OR z=0 then
         xx = x-x
@@ -1472,24 +1472,24 @@
         ExtentRange3D[\I,J,K\](x,y,z,1,1,1)
     end
 
-right1Range[\I extends AnyIntegral\](_:I,x:I):RightRange[\I\] = RightScalarRange[\I\](x,1)
-right2Range[\I extends AnyIntegral, J extends AnyIntegral\](_:I,_:J,x:I,y:J):
+right1Range[\I extends Integral[\I\]\](_:I,x:I):RightRange[\I\] = RightScalarRange[\I\](x,1)
+right2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_:I,_:J,x:I,y:J):
     RightRange[\(I,J)\] = RightRange2D[\I,J\](x,y,1,1)
-right3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+right3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
            (_:I,_:J,_:K,x:I,y:J,z:K): RightRange[\(I,J,K)\] =
     RightRange3D[\I,J,K\](x,y,z,1,1,1)
 
-openRangeHelper[\I extends AnyIntegral\](_ : ()->I):
+openRangeHelper[\I extends Integral[\I\]\](_ : ()->I):
     OpenScalarRange[\I\] = OpenScalarRange[\I\](1)
-openRangeHelper[\I extends AnyIntegral, J extends AnyIntegral\](_ : ()->(I,J)):
+openRangeHelper[\I extends Integral[\I\], J extends Integral[\J\]\](_ : ()->(I,J)):
     OpenRange2D[\I,J\] = OpenRange2D[\I\](1,1)
-openRangeHelper[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+openRangeHelper[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
                (_ : ()->(I,J,K)): OpenRange3D[\I,J,K\] = OpenRange3D[\I,J,K\](1,1,1)
 
-open1Range[\I extends AnyIntegral\](_:I,x:I):OpenRange[\I\] = OpenScalarRange[\I\](x)
-open2Range[\I extends AnyIntegral, J extends AnyIntegral\](_:I,_:J,x:I,y:J):
+open1Range[\I extends Integral[\I\]\](_:I,x:I):OpenRange[\I\] = OpenScalarRange[\I\](x)
+open2Range[\I extends Integral[\I\], J extends Integral[\J\]\](_:I,_:J,x:I,y:J):
     OpenRange[\(I,J)\] = OpenRange2D[\I,J\](x,y)
-open3Range[\I extends AnyIntegral, J extends AnyIntegral, K extends AnyIntegral\]
+open3Range[\I extends Integral[\I\], J extends Integral[\J\], K extends Integral[\K\]\]
            (_:I,_:J,_:K,x:I,y:J,z:K): OpenRange[\(I,J,K)\] =
     OpenRange3D[\I,J,K\](x,y,z)
 
