--- a/ProjectFortress/src/com/sun/fortress/interpreter/Driver.java	2026-09-29 14:24:46.717929654 +0000
+++ b/ProjectFortress/src/com/sun/fortress/interpreter/Driver.java	2026-09-29 14:25:50.759080925 +0000
@@ -462,8 +462,8 @@
             List<Import> imports = newapi.getImports();
             ensureImportsImplemented(fr, linker, pile, imports);
         }
-        // TODO temp hack till we knit in natives properly.
-        return (ComponentWrapper) newwrapper;
+        // A foreign api is implemented by a ForeignComponentWrapper, not a ComponentWrapper.
+        return newwrapper instanceof ComponentWrapper ? (ComponentWrapper) newwrapper : null;
     }
 
     private static ComponentWrapper commandLineComponent(GraphRepository fr,
