diff --git a/Library/FortressLibrary.fsi b/Library/FortressLibrary.fsi
index c48db16f6..45f2e87b0 100644
--- a/Library/FortressLibrary.fsi
+++ b/Library/FortressLibrary.fsi
@@ -1580,34 +1580,34 @@ vector[\T extends Number, nat s0\](v:T):Vector[\T,s0\]
 tabulatedVector[\T extends Number, nat s0\](f:ZZ32->T):Vector[\T,s0\]
 
 
-opr +[\ T extends Number, nat n, nat m \]
+opr +[\ T extends Number, nat n \]
      (me : Vector[\T,n\], other : Vector[\T,n\]):Vector[\T,n\]
 
-opr -[\ T extends Number, nat n, nat m \]
+opr -[\ T extends Number, nat n \]
      (me : Vector[\T,n\], other : Vector[\T,n\]):Vector[\T,n\]
 
-opr -[\ T extends Number, nat n, nat m \]
+opr -[\ T extends Number, nat n \]
      (me : Vector[\T,n\]):Vector[\T,n\]
 
 pmul[\ T extends Number, nat k \]
     (a : Vector[\T,k\], b : Vector[\T,k\]):Vector[\T,k\]
 
-opr DOT[\ T extends Number, nat n, nat m, nat p \]
+opr DOT[\ T extends Number, nat n \]
        (me : Vector[\T,n\], other : Vector[\T,n\]):T
 
-opr juxtaposition[\ T extends Number, nat n, nat m, nat p \]
+opr juxtaposition[\ T extends Number, nat n \]
      (me : Vector[\T,n\], other : Vector[\T,n\]):T
 
-opr DOT[\ T extends Number, nat n, nat m, nat p \]
+opr DOT[\ T extends Number, nat n \]
        (me : Vector[\T,n\], other : T) : Vector[\T,n\]
 
-opr juxtaposition[\ T extends Number, nat n, nat m, nat p \]
+opr juxtaposition[\ T extends Number, nat n \]
      (me : Vector[\T,n\], other : T) : Vector[\T,n\]
 
-opr DOT[\ T extends Number, nat n, nat m, nat p \]
+opr DOT[\ T extends Number, nat n \]
         (other : T, me : Vector[\T,n\]) : Vector[\T,n\]
 
-opr juxtaposition[\ T extends Number, nat n, nat m, nat p \]
+opr juxtaposition[\ T extends Number, nat n \]
      (other : T, me : Vector[\T,n\]) : Vector[\T,n\]
 
 squaredNorm[\T extends Number, nat s0\](a:Vector[\T,s0\]):T
@@ -1705,29 +1705,29 @@ opr juxtaposition[\ T extends Number, nat n, nat m, nat p\]
      (me:Matrix[\T,n,m\], other:Matrix[\T,m,p\]): Matrix[\T,n,p\]
 
 (** Matrix-vector multiplication. *)
-opr DOT[\ T extends Number, nat n, nat m, nat p \]
+opr DOT[\ T extends Number, nat n, nat m \]
        (me:Matrix[\T,n,m\], v:Vector[\T,m\]):Vector[\T,n\]
 
-opr juxtaposition[\ T extends Number, nat n, nat m, nat p \]
+opr juxtaposition[\ T extends Number, nat n, nat m \]
      (me:Matrix[\T,n,m\], v:Vector[\T,m\]):Vector[\T,n\]
 
 (** Vector-matrix multiplication. *)
-opr DOT[\ T extends Number, nat n, nat m, nat p \]
+opr DOT[\ T extends Number, nat n, nat m \]
        (v:Vector[\T,n\], me:Matrix[\T,n,m\]):Vector[\T,m\]
 
-opr juxtaposition[\ T extends Number, nat n, nat m, nat p \]
+opr juxtaposition[\ T extends Number, nat n, nat m \]
      (v:Vector[\T,n\], me:Matrix[\T,n,m\]):Vector[\T,m\]
 
-opr DOT[\ T extends Number, nat n, nat m, nat p \]
+opr DOT[\ T extends Number, nat n, nat m \]
        (me : Matrix[\T,n,m\], other : T) : Matrix[\T,n,m\]
 
-opr juxtaposition[\ T extends Number, nat n, nat m, nat p \]
+opr juxtaposition[\ T extends Number, nat n, nat m \]
      (me : Matrix[\T,n,m\], other : T) : Matrix[\T,n,m\]
 
-opr DOT[\ T extends Number, nat n, nat m, nat p \]
+opr DOT[\ T extends Number, nat n, nat m \]
        (other : T, me : Matrix[\T,n,m\]) : Matrix[\T,n,m\]
 
-opr juxtaposition[\ T extends Number, nat n, nat m, nat p \]
+opr juxtaposition[\ T extends Number, nat n, nat m \]
      (other : T, me : Matrix[\T,n,m\]) : Matrix[\T,n,m\]
 
 (** %Array3[\T,b0,s0,b1,s1,b2,s2\]% is the type of 3-dimensional arrays
