diff --git a/ProjectFortress/LibraryBuiltin/CompilerBuiltin.fss b/ProjectFortress/LibraryBuiltin/CompilerBuiltin.fss
index 7df782075..1d3128c20 100644
--- a/ProjectFortress/LibraryBuiltin/CompilerBuiltin.fss
+++ b/ProjectFortress/LibraryBuiltin/CompilerBuiltin.fss
@@ -991,6 +991,7 @@ end
 
 trait RR32 extends { Number, Equality[\RR32\] } excludes { ZZ64, ZZ32, RR64 }
     coerce(x: FloatLiteral) = x.asRR32
+    opr =(self, other:RR32): Boolean = jFloatToDouble(self) = jFloatToDouble(other)
 end
 
 trait FloatLiteral excludes {RR32, RR64}
