--- a/Library/FortressLibrary.fss
+++ b/Library/FortressLibrary.fss
@@ -379,7 +379,7 @@
         z: NN32 => do q: QQ = z; q end
         z: NN64 => do q: QQ = z; q end
         z: ZZ => do q: QQ = z; q end
-        else => Ratio(0, 0)
+        else => do printlnWithThread("PROBEQ-EXACTVALUE-ELSE"); Ratio(0, 0) end
     end
 
 trait RR64 extends { Number, StandardPartialOrder[\RR64\], StandardMinMax[\RR64\],
